actions/class.XmlEditor.php

```diff
@@ -35,18 +36,31 @@ public function edit() : void
             return;
        }
        $test = $this->getResource($this->getPostParameter('id'));
+       $title = __('XML Content');
 
        if ($this->getXmlEditorService()->isLocked()) {
-           $this->setData('lockMessage', __('This functionality is blocked. Please contact with your administrator for more details.'));
-           $this->setData('isLocked', true);
+           $title = __('This functionality is blocked. Please contact with your administrator for more details.');
        } else {
            try {
                $xmlString = $this->getXmlEditorService()->getTestXml($test);
+
+               $formContainer = new taoQtiTest_models_forms_XmlEditForm(
+                   $test,
+                   $xmlString,
+                   [FormContainer::CSRF_PROTECTION_OPTION => true]
+               );
+               $form = $formContainer->getForm();
+               if ($form->isSubmited() && $form->isValid()) {
+                   $this->getXmlEditorService()->saveStringTest($test, $form->getValues()['xmlString']);
+                   $this->setData('message', __('Saved'));
+               }
+               $this->setData('form', $form->render());
+
            } catch (\Exception $e) {
-               $xmlString = $e->getMessage();
+               $title = $e->getMessage();
```

Comment:
Not sure that this would be this way, until we are in debug mode - are you sure we want to disclose all internal errors raised on client level (since `catch \Exception`) ?